internal/worker/client_test.go

```diff
@@ -128,12 +128,11 @@ func TestProxy(t *testing.T) {
 	require.False(t, c)
 	require.NoError(t, err)
 
-	// we expect 6 calls to go through the proxy:
-	// - register worker
+	// we expect 5 calls to go through the proxy:
 	// - request job (fails, no oauth token)
 	// - oauth call
 	// - request job (succeeds)
 	// - upload artifact
 	// - cancel
-	require.Equal(t, 6, proxy.calls)
+	require.Equal(t, 5, proxy.calls, "We got those:\n"+strings.Join(proxy.paths, "\n"))
```

Comment:
Can we somehow test that the client still registers in a unit test?

Reply:
I would have registered only after 1 Minute. So I'll revise the implementation.

Reply:
Ahhh ic, yea no that's ok then, no point waiting a minute in a unit test.

Reply:
I'll suggest a more straight forward implementation. (already started implementing)
It should not be the "usual behavior" to start the interaction with a failed request due to missing authentication in the first place ;)